### ISSUE TITLE
Improved cover import

### ISSUE THREAD

**Issue by aristippe** [aristippe/pathagar#26]
Many ePubs lack specifying  "cover" as an image. In those cases, cover references a html file:

```
 <guide>
     <reference href="cover.xhtml" title="Cover" type="cover"/>
 </guide>

```

The proportion of such ePubs is fairly high and those end up without a cover. I think it's worth improving the cover import. Perhaps Calibre renders the HTML as an image, though it often does so incorrectly with improper scaling and positioning. Maybe extracting the cover HTML, and then taking any image in it as the cover, or as a first attempt, rendering it as an image, needed in cases of SVG, is a good approach.

**Comment by sinergatis** [comment]
Makes much sense, and thanks for looking into the specification and for the information on how common this issue is.

I like the idea of trying the a simple "_if cover is html, use the first/biggest/only image on it approach_" solution first, and checking if this solves acceptably a significant percentage of epubs. Rendering the HTML sounds good too, but looks rather complex and involving quite a few components - we might end up finding it easier replacing the current epub parser with a third-party one, which might bring some other benefits when parsing as well.

**Comment by aristippe** [comment]
I think got I most of it in PR 34 for the epub-cover branch. For when the cover was xhtml, I took the first img tag, or image, which happens with it's embedded in an svg tag. Tested with about 800 epubs and checking a few cases where there was no cover, there was no cover specified. If we want to work around it more, it's possible to look at the first file in the spine, to see if it has cover, or to render an svg string as an image.

There doesn't seem to be many epub parsers that are well maintained. The one from Calibre looks interesting, at least for adapting some code.
